include/zephyr/devicetree/spi.h

```diff
@@ -186,7 +189,7 @@ extern "C" {
  * @return label property of spi_dev's chip select GPIO controller
  */
 #define DT_SPI_DEV_CS_GPIOS_LABEL(spi_dev) \
-	DT_GPIO_LABEL_BY_IDX(DT_BUS(spi_dev), cs_gpios, DT_REG_ADDR(spi_dev))
+	DT_GPIO_LABEL_BY_IDX(DT_BUS(spi_dev), cs_gpios, DT_REG_ADDR(spi_dev)) __DEPRECATED_MACRO
```

Comment:
devicetree.h is stable, and I think deprecations should be treated as incompatible changes, so this should go through the stable API change process: https://docs.zephyrproject.org/latest/develop/api/api_lifecycle.html#introducing-incompatible-changes

I don't see an RFC issue or email to devel about this, so the architecture meeting hasn't had a chance to review this either.

It's important to keep users in the loop about changes like these by following the process. NAK until that's done.

Reply:
@mbolivar-nordic You do not need to go through the stable API change procedure if you deprecate. The whole point of that procedure is to skip deprecation altogether. Deprecating an API does not require anything beyond the Pull Request.

Reply:
Maybe this should be pointed out more clearly in the API lifecycle doc, but this is how we have always done this.